### PR TITLE
add opencensus handler to shuttle

### PR DESCRIPTION
**Pull request by lanzafame**
Also, add interface compile-time check for rcmgr.MetricsReporter.

### CODE REVIEW

**Comment by lanzafame** [comment]
The extra .gitignore entries are based of the what estuary/shuttle add to the directory when I started them with minimal flags.

**Review by alvin-reyes — APPROVED** [review]
LGTM

**Comment by gmelodie** [comment]
> is the opencensus endpoint actually different than the prometheus one?

@whyrusleeping it doesn't seem to be different in this case. [Exporter()](https://github.com/lanzafame/estuary/blob/401b92946f194b4a00d7dc2d7b6972a5721cd7d7/metrics/exporter.go#L11-L29) basically creates a default [Exporter](https://github.com/census-ecosystem/opencensus-go-exporter-prometheus/blob/aaaf21250a48ed3f2da42bab1aab23ae7d48bcd6/prometheus.go#L36-L41) object with the `estuary` namespace. But in the end it acts as an API endpoint it seems to only return the `http.Handler` part of that object (not sure how Go is clever to make that work, maybe the casting in the function return?). It seems that `Exporter()` and `promhttp.Handler()` return almost identical objects the way we have `Exporter()` implemented now, that is. I think the way to go is to use `Exporter()` so that we can add options and configs if need be in the future.

File: cmd/estuary-shuttle/main.go [hunk 1]
```diff
@@ -376,6 +376,7 @@ func main() {
 		})
 
 		go func() {
+			http.Handle("/debug/metrics/opencensus", estumetrics.Exporter())
 			http.Handle("/debug/metrics/prometheus", promhttp.Handler())
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:380]: We may want to remove this endpoint in favor of the new one

**Review by gmelodie — COMMENTED** [review]
Small remark, but looking good!

**Comment by lanzafame** [comment]
@whyrusleeping @gmelodie personal opinion, we remove the opencensus and prometheus endpoints and just have `/debug/metrics`. Thoughts?

**Comment by lanzafame** [comment]
@gmelodie Shall I make the same endpoint change to the main estuary handlers as well?

**Pull request merged**